docs/rfc/rfc-9.adoc

```diff
@@ -0,0 +1,145 @@
+https://www.apmex.com/product/117942/1932-s-washington-quarter-choice-au[special
+edition quarters]. The the linked quarter trades for ~$350, but is also legal
+tender worth $0.25. An owner of such a quarter probably does not want to use it
+to pay for parking. Inscriptions work the same way.
+
+Since Ordinals are still BTC, we can use most of the same tBTCv2 ideas for
+custodying Ordinals: rotating wallets, P2SH addresses, fraud proofs,
+redemptions, etc, with some slight modifications.
+
+== Proposal
+
+=== Overview
+
+We build a BTC <> Ethereum Ordinal bridge!
+
+We create 51-of-100 threshold signing groups using the same process that we use
+for tBTCv2. To deposit to these signing groups, we use the same P2SH process
+including a refund that we use for tBTCv2. They pay a <<Fees,fee>>. We
+optimistically mint using the same Minter & Guardian process we already have. In
+the background, we sweep to disable the refund. Later, someone brings back the
+NFT and a <<Fees,fee>>. We burn the NFT and send them their Ordinal.
+
+=== Depositing Details
+
+Depositing works essentially the same way that it does in tBTCv2 with a few key
+differences.
+
+Once a deposit is revealed, it is **extremely** important that the wallet does
+not sweep them into the same UTXO; that would be destructive. We still need to
+sweep to disable the refundfootnote:[Whether or not the sweep can be performed
+in a batch is a further research item.]. The sweep needs to preserve the
+individual satoshis.
+
+Once custodied, we would mint an ERC721 where the `metadata` field references
+the https://docs.ordinals.com/overview.html[Ordinal ID]. They offer multiple
+notations; using the Integer Notation looks simple. The reference to the Ordinal
+ID is enough for a dApp to pull the underlying content, but it needs to be
+plugged in to both Bitcoin and Ethereum.
+
+The ERC721 in the user's Ethereum Wallet points at the Ordinal ID. Then, the
+https://github.com/casey/ord[ord] program can identify the associated Ordinal on
+BTC, and extract its payload, which looks like:
+
+```
+OP_FALSE
+OP_IF
+  OP_PUSH "ord"
+  OP_1
+  OP_PUSH "<content-type>" // example: OP_PUSH "text/plain;charset=utf-8"
+  OP_0
+  OP_PUSH "<content>" // example: OP_PUSH "Hello, world!"
+OP_ENDIF
+```
+
+=== Redemption Details
+
+Since NFTs are unique, there is no longer choice about which wallet a NFT is
+redeemed from. It **must** be the wallet that is the custody of that Ordinal.
+
+The redemption process is otherwise identical to tBTCv2. Someone sends the NFT,
+a <<Fees,fee>>, and an ethereum address back to contracts. The associated wallet
+sends them their Ordinal to the provided address with a 51-of-100 signature.
+
+=== Fraud Detection
+
+Rather than storing the full list of custodied Ordinals per
+walletfootnote:[Storing the full list would of Ordinals be very gas
+inefficient.], we instead:
+
+* Take the merkle root of all Ordinals under custody
```

Comment:
When does this get updated? By whom?

Reply:
> Since NFTs are unique, there is no longer a choice about which wallet an NFT is
redeemed from. It **must** be the wallet that is the custody of that Ordinal.

I may need help understanding this. Do you mean I cannot bridge back the Ordinal unless I own the wallet where the Ordinal is kept?

This means that the UI will be a smidge affected; right now, we give the liberty of the tBTC owner to send their BTC back to any address they want. 

If I understand correctly, what you are saying means that when the Ordinal ERC721 token owner wants to bridge back, the dApp will display the BTC Wallet address where the Ordinal will be sent back in a purely informative way. The input label for the BTC Wallet will be disabled and not editable.

Reply:
> I may need help understanding this. Do you mean I cannot bridge back the Ordinal unless I own the wallet where the Ordinal is kept?

### BTC
Say that you bridge 10 BTC into tBTCv2 during the first week of the system. The first wallet will custody your bitcoin, and you'll get 10 tBTC. Then, a few months pass and now there are 7 alive wallets. The oldest has already been redeemed for all of its funds, so the original you deposited to is gone.

You go to redeem. The dApp will request a redemption from the oldest wallet, but from the system's perspective and in terms of fungibility, it doesn't matter. You're going to be given 10 BTC from *any* wallet, because we assume you don't care which 10 BTC you're getting - they're all the same (like barrels of oil).

### Ordinals
If you deposit an Ordinal into the system and get a NFT for that Ordinal, when you redeem that NFT you need to get exactly that same Ordinal back. That means that if there are 8 alive wallets (each the custody of many unique Ordinals), then **only one** of those Wallets has **your** Ordinal.

Ordinal-NFT holders on Ethereum don't own any of the BTC wallets that custody the Ordinals; that's the job of the node operators.